source/core_mqtt.c

```diff
@@ -1796,7 +1796,7 @@ MQTTStatus_t MQTT_Connect( MQTTContext_t * pContext,
                                 pContext->networkBuffer.pBuffer,
                                 packetSize );
 
-        if( bytesSent < 0 )
+        if( bytesSent < ( int32_t ) packetSize )
```

Comment:
This test collapses both error conditions (`bytesSent < 0` and `bytesSent < packetSize`), so the error message on line 1801 is not as helpful as it should be.  If `bytesSent < 0` the actual error returned from the underlying transport is discarded – this can be very important when debugging so it should be logged.

It looks like these error messages had the same problem in the previous revision.  Let's improve them to make it easier to debug networking issues.

Same comment at 1890, 2023, 2085, & 2135.

Reply:
For a negative value error code returned by the `Transport_Send_t` call, the error code is logged in the `sendPacket` function here: https://github.com/FreeRTOS/coreMQTT/blob/main/source/core_mqtt.c#L625
Thus, as long as the logging is enabled for the library at `LOG_ERROR` level, the return code from the networking stack will be logged.

Also, if there was a timeout in sending the packet, that will also be logged separately by the function.

Reply:
What if it's HTTP?